extensions-core/azure-extensions/src/main/java/org/apache/druid/storage/azure/AzureDataSegmentPuller.java

```diff
@@ -59,7 +64,7 @@ FileUtils.FileCopyResult getSegmentFiles(
           "Loading container: [%s], with blobPath: [%s] and outDir: [%s]", containerName, blobPath, outDir
       );
 
-      final String actualBlobPath = AzureUtils.maybeRemoveAzurePathPrefix(blobPath);
+      final String actualBlobPath = azureUtils.maybeRemoveAzurePathPrefix(blobPath);
```

Comment:
you could pass the endpoint suffix as an argument than passing it in the AzureUtils constructor

Reply:
Updated